<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>920</width>
    <height>647</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Serial Terminal</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QVBoxLayout" name="mainVerticalLayout">
      <item>
       <widget class="QGroupBox" name="groupSerial">
        <property name="title">
         <string>Serial Settings</string>
        </property>
        <layout class="QHBoxLayout" name="layoutSerial">
         <item>
          <widget class="QLabel" name="labelPort">
           <property name="text">
            <string>Port:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboPort"/>
         </item>
         <item>
          <widget class="QPushButton" name="btnRefresh">
           <property name="text">
            <string>üîÑ Refresh</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="labelBaud">
           <property name="text">
            <string>Baud:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboBaud">
           <item>
            <property name="text">
             <string>110</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>300</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>600</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>1200</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>2400</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>4800</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>9600</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>14400</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>19200</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>28800</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>38400</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>57600</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>115200</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>128000</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>230400</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>250000</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>256000</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>460800</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>500000</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>921600</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Custom...</string>
            </property>
           </item>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboFormat">
           <item>
            <property name="text">
             <string>8N1</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>8N2</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>8E1</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>8O1</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>8M1</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>8S1</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>7E1</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>7O1</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>7E2</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>7O2</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>6N1</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>5N1</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>5N1.5</string>
            </property>
           </item>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btnConnect">
           <property name="text">
            <string>üîå Connect</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btnDisconnect">
           <property name="text">
            <string>‚ùå Disconnect</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupOutput">
        <property name="title">
         <string>Serial Output</string>
        </property>
        <layout class="QVBoxLayout" name="layoutOutput">
         <item>
          <widget class="QTextEdit" name="textOutput">
           <property name="font">
            <font>
             <family>Courier</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="horizontalScrollBarPolicy">
            <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
           </property>
           <property name="lineWrapMode">
            <enum>QTextEdit::LineWrapMode::WidgetWidth</enum>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupSend">
        <property name="title">
         <string>Send Command</string>
        </property>
        <layout class="QHBoxLayout" name="layoutSend">
         <item>
          <widget class="QLabel" name="labelSend">
           <property name="text">
            <string>Send:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineInput"/>
         </item>
         <item>
          <widget class="QPushButton" name="btnSend">
           <property name="text">
            <string>Send</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="labelEOL">
           <property name="text">
            <string>EOL:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboEOL">
           <item>
            <property name="text">
             <string>None</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>NL (\n)</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>CR (\r)</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Both (\r\n)</string>
            </property>
           </item>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="checkTimestamp">
           <property name="text">
            <string>Timestamp</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="checkDarkTheme">
           <property name="text">
            <string>Dark Mode</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="layoutControls">
        <item>
         <widget class="QCheckBox" name="checkPause">
          <property name="text">
           <string>Pause Printing</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnClearLog">
          <property name="text">
           <string>üßπ Clear Log</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="spacerControls">
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Policy::Expanding</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QCheckBox" name="checkAutoScroll">
          <property name="text">
           <string>Auto-scroll</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnSaveLog">
          <property name="text">
           <string>üíæ Save Log</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QGroupBox" name="groupHistory">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>200</width>
        <height>600</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>200</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>Command History</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayoutHistory">
       <item>
        <widget class="QListWidget" name="listHistory"/>
       </item>
	   <item>
		  <widget class="QPushButton" name="btnSortHistory">
			<property name="text">
			  <string>Sort A‚ÄìZ</string>
			</property>
			<property name="toolTip">
			  <string>Sort commands alphabetically (A‚ÄìZ)</string>
			</property>
		  </widget>
		</item>
       <item>
        <widget class="QPushButton" name="btnClearHistory">
         <property name="text">
          <string>üßπ Clear History</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar">
   <widget class="QLabel" name="labelStatusLED">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>12</width>
      <height>12</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>12</width>
      <height>12</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>12</width>
      <height>12</height>
     </size>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: #888; border-radius: 6px;</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelStatusText">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>100</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Ready</string>
    </property>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
